src/Symfony/Component/VarDumper/Dumper/ServerDumper.php

```diff
@@ -42,6 +50,42 @@ public function __construct(string $host, DataDumperInterface $wrappedDumper = n
         $this->contextProviders = $contextProviders;
     }
 
+    /**
+     * @final
+     */
+    public static function getDefaultContextProviders(Request $request = null, string $projectDir = null): array
+    {
+        $contextProviders = array();
+
+        if ('cli' !== PHP_SAPI && ($request || class_exists(Request::class))) {
+            $requestStack = new RequestStack();
+            $requestStack->push($request ?? Request::createFromGlobals());
+            $contextProviders['request'] = new RequestContextProvider($requestStack);
+        }
+
+        $fileLinkFormatter = class_exists(FileLinkFormatter::class) ? new FileLinkFormatter(null, $requestStack ?? null, $projectDir) : null;
+
+        return $contextProviders + array(
+            'cli' => new CliContextProvider(),
+            'source' => new SourceContextProvider(null, $projectDir, $fileLinkFormatter),
+        );
+    }
+
+    /**
+     * @final
+     */
+    public static function register(string $host = null, array $contextProviders = array()): ?callable
+    {
+        $contextProviders += self::getDefaultContextProviders();
```

Comment:
Having getDefaultContextProviders called with self:: and with no argument here + that the method is final means there is no way to override the default context providers here. This might be fine (is it?) but then, what's the prupose of a standalone getDefaultContextProviders method?